nilearn/datasets/utils.py

```diff
@@ -154,12 +159,17 @@ def _chunk_read_(response, local_file, chunk_size=8192, report_hook=None,
     bytes_so_far = initial_size
 
     t0 = time.time()
+    _chunk_period = total_size // (chunk_size * report_frequency)
+    _block_size = initial_size + chunk_size
     while True:
         chunk = response.read(chunk_size)
         bytes_so_far += len(chunk)
-        if report_hook:
-            _chunk_report_(len(chunk), bytes_so_far, total_size, initial_size, t0)
 
+        # Reporting download progress by block
+        if report_hook and bytes_so_far == _block_size:
```

Comment:
I think that I would prefer having an iteration counter of the "while" loop, and reporting only if that iteration counter is an integer fraction of report_frequency, ie
<pre>
if i // report_frequency == 1:
   report
</pre>

I find that such a logic is more robust than a logic on the chunk size (and easier to read).